semgrep-core/src/parsing/tree_sitter/Parse_cpp_tree_sitter.ml

```diff
@@ -1276,66 +1329,69 @@ and map_compound_literal_expression (env : env)
       let v2 = map_type_descriptor env v2 in
       let v3 = token env v3 (* ")" *) in
       let v4 = map_initializer_list env v4 in
-      todo env (v1, v2, v3, v4)
+      GccConstructor ((v1, v2, v3), v4)
   | `Choice_id_init_list (v1, v2) ->
       let v1 =
         match v1 with
         | `Id tok ->
             let x = str env tok (* pattern [a-zA-Z_]\w* *) in
-            todo env x
+            complicated env x
         | `Temp_type x ->
             let x = map_template_type env x in
-            todo env x
+            complicated env x
         | `Scoped_type_id x ->
             let x = map_scoped_type_identifier env x in
-            todo env x
+            complicated env x
       in
-      let v2 = map_initializer_list env v2 in
-      todo env (v1, v2)
+      let l, xs, r = map_initializer_list env v2 in
+      complicated env (v1, v2)
 
 and map_compound_statement (env : env) ((v1, v2, v3) : CST.compound_statement) :
     compound =
   let v1 = token env v1 (* "{" *) in
   let v2 = map_translation_unit env v2 in
   let v3 = token env v3 (* "}" *) in
-  todo env (v1, v2, v3)
+  complicated env (v1, v2, v3)
 
 and map_condition_clause (env : env) ((v1, v2, v3) : CST.condition_clause) :
     condition_clause paren =
   let v1 = token env v1 (* "(" *) in
   let v2 =
     match v2 with
-    | `Opt_choice_decl_choice_exp (v1, v2) ->
-        let _v1TODO =
+    | `Opt_choice_decl_choice_exp (v1, v2) -> (
+        let v1 =
           match v1 with
           | Some x -> (
               match x with
-              | `Decl x -> map_declaration env x
+              | `Decl x -> Common2.Left3 (map_declaration env x)
               | `Exp_stmt x ->
                   let eopt, sc = map_expression_statement env x in
-                  todo env x)
-          | None -> todo env ()
+                  Common2.Right3 (eopt, sc))
+          | None -> Common2.Middle3 ()
         in
         let v2 = map_anon_choice_exp_55b4dba env v2 in
-        CondClassic v2
+        match v1 with
+        | Common2.Middle3 () -> CondClassic v2
+        | Common2.Left3 d -> complicated env d
+        | Common2.Right3 e -> complicated env e)
     | `Cond_decl x -> map_condition_declaration env x
   in
   let v3 = token env v3 (* ")" *) in
   (v1, v2, v3)
 
 and map_condition_declaration (env : env)
     ((v1, v2, v3) : CST.condition_declaration) =
-  let v1 = map_declaration_specifiers env v1 in
+  let t, specs = map_declaration_specifiers env v1 in
   let v2 = map_declarator env v2 in
   let v3 =
     match v3 with
     | `EQ_exp (v1, v2) ->
         let v1 = token env v1 (* "=" *) in
         let v2 = map_expression env v2 in
-        todo env (v1, v2)
+        complicated env (v1, v2)
     | `Init_list x -> map_initializer_list env x
   in
-  todo env (v1, v2, v3)
+  complicated env (v1, v2, v3)
```

Comment:
Please try to give names to the `v1`, `v2`, ... variables if you know what they are. Not having names makes any later code modification really harder than it should be.